### PR TITLE
 Illuminate\Support\Collection support

### PR DESCRIPTION
**Pull request by zbrag**
Made it possible to use Illuminate\Support\Collection instead of only Illuminiate\Database\Eloquent\Collection.
Also added a test to see if my patch works

### CODE REVIEW

**Comment by usmanhalalit** [comment]
Thanks @zbrag! Can you please give me some use cases where Collection export will be useful?

File: src/Laracsv/Export.php [hunk 1]
```diff
@@ -101,14 +101,18 @@ public function getCsv()
...
-     * @param \Illuminate\Database\Eloquent\Collection $collection
+     * @param \Illuminate\Support\Collection $collection
      * @param array $fields
      * @param \League\Csv\Writer $csv
      * @return void
      */
     private function addCsvRows(Collection $collection, array $fields, Writer $csv)
     {
-        $collection->makeVisible($fields);
+        $isEloquentCollection = false;
```
Comment (max-kovpak) [inline src/Laracsv/Export.php]: $isEloquentCollection = is_a($collection, \Illuminate\Database\Eloquent\Collection::class);
if ($isEloquentCollection) {
    $collection->makeVisible($fields);
}

**Comment by max-kovpak** [comment]
@usmanhalalit for example, if you need to create CSV using some custom data gathered from different models or calculated in the repositories.

**Comment by usmanhalalit** [comment]
Closing as it does not seem as useful.

**Comment by s512** [comment]
@usmanhalalit this feature would be useful for us too!

In some cases we're using the DB facade to fetch data (because of complex joins), and this returns an Illuminate\Support\Collection object. For example: `DB::table('users')->get();`

But we're not able to export this data using this package, as it expects Illuminate\Database\Eloquent\Collection.

**Comment by usmanhalalit** [comment]
Thanks @s512, reopened.

**Comment by Pablodotnet** [comment]
When does this is going to be able on the branch you say on readme to install?

**Comment by usmanhalalit** [comment]
@Pablodotnet Please use `composer require "usmanhalalit/laracsv:dev-master"` to have this for now.

**Pull request merged**